### ISSUE TITLE
[OpenMP] BF16 chokes offload to amdgpu

### ISSUE THREAD

**Issue by ye-luo** [llvm/llvm-project#59253]
Reproducer
```cpp
#include <emmintrin.h>
int main()
{}
```
clang error
```console
yeluo@epyc-server:~/temp$ clang++ -fopenmp --offload-arch=sm_61 bf16_err.cpp 
yeluo@epyc-server:~/temp$ clang++ -fopenmp --offload-arch=gfx906 bf16_err.cpp 
In file included from fp64_err.cpp:1:
/soft/llvm/main-patched/lib/clang/16/include/emmintrin.h:47:9: error: __bf16 is not supported on this target
typedef __bf16 __v8bf __attribute__((__vector_size__(16), __aligned__(16)));
        ^
/soft/llvm/main-patched/lib/clang/16/include/emmintrin.h:47:38: error: vector size not an integral multiple of component size
typedef __bf16 __v8bf __attribute__((__vector_size__(16), __aligned__(16)));
                                     ^               ~~
/soft/llvm/main-patched/lib/clang/16/include/emmintrin.h:48:9: error: __bf16 is not supported on this target
typedef __bf16 __m128bh __attribute__((__vector_size__(16), __aligned__(16)));
        ^
/soft/llvm/main-patched/lib/clang/16/include/emmintrin.h:48:40: error: vector size not an integral multiple of component size
typedef __bf16 __m128bh __attribute__((__vector_size__(16), __aligned__(16)));
```
Caused by [e0fb01e97b6b7d2fe66b17b36eeb98aa78c6e3bb](https://reviews.llvm.org/rGe0fb01e97b6b7d2fe66b17b36eeb98aa78c6e3bb) 
The issue was worked around in HIP. https://reviews.llvm.org/D138651

**Comment by ye-luo** [comment]
> Might be fixed with https://reviews.llvm.org/D138651

I referenced this patch. No luck.
It doesn't help openmp, probably need to do something similar.

**Comment by ye-luo** [comment]
Fixed by 678d8946ba2ba790c4c52e96e2134ee136e30057
